File: TransactionRegister/Model/Client/Amount.m
//
//  Amount.m
//  TransactionRegister
//
//  Created by Eric Romrell on 9/4/16.
//  Copyright © 2016 Eric Romrell. All rights reserved.
//

#import "Amount.h"

@implementation Amount

+(Amount *)amountWithDouble:(double)value {
	Amount *me = [[Amount alloc] init];
	me.value = value;
	return me;
}

-(NSString *)formattedValue {
	NSNumberFormatter *format = [NSNumberFormatter new];
	[format setNumberStyle:NSNumberFormatterCurrencyStyle];
	return [format stringFromNumber:[NSNumber numberWithDouble:self.value]];
}

@end

File: TransactionRegister/Model/Client/Amount.swift
//
//  NewAmount.swift
//  TransactionRegister
//
//  Created by Eric Romrell on 1/18/17.
//  Copyright © 2017 Eric Romrell. All rights reserved.
//

import Foundation

class Amount : NSObject {
	var value : Double
	
	init(value: Double) {
		self.value = value;
	}
	
	func formattedValue() -> String? {
		let format = NumberFormatter()
		format.numberStyle = .currency
		return format.string(from: NSNumber(floatLiteral: self.value))
	}
}
